src/api/requests.ts

```diff
@@ -34,73 +26,77 @@ export const get = async <T>(
 export const post = async <T>(
   url: string,
   data: any,
-  auth = false,
   content_type = 'application/json'
 ) => {
   const request = new Request(baseUrl + url, {
     ...(content_type === 'application/json' && {
       headers: {
         'Content-Type': 'application/json',
-        'Access-Control-Allow-Origin': '*',
       },
       body: JSON.stringify(data),
     }),
+    credentials: 'include',
     method: 'POST',
     ...(content_type === 'multipart/form-data' && { body: data }),
   });
 
-  if (auth) {
-    return authFetch<T>(request);
-  }
-  return fetch(request).then((res) => handleResponse<T>(res));
+  return authFetch<T>(request);
 };
 
-export const put = async <T>(url: string, data: any, auth = false) => {
+export const put = async <T>(url: string, data: any) => {
   const request = new Request(baseUrl + url, {
     method: 'PUT',
     body: JSON.stringify(data),
     headers: {
       'content-type': 'application/json',
-      'Access-Control-Allow-Origin': '*',
     },
+    credentials: 'include',
   });
-  if (auth) {
-    return authFetch<T>(request);
-  }
-  return fetch(request).then((res) => handleResponse<T>(res));
+  return authFetch<T>(request);
 };
 
-export const Delete = async <T>(url: string, auth = false) => {
+export const Delete = async <T>(url: string) => {
   const request = new Request(baseUrl + url, {
     method: 'DELETE',
+    credentials: 'include',
     headers: {
       'content-type': 'application/json',
-      'Access-Control-Allow-Origin': '*',
     },
   });
-  if (auth) {
-    return authFetch<T>(request);
-  }
-  return fetch(request).then((res) => handleResponse<T>(res));
+  return authFetch<T>(request);
 };
 
-const authFetch = <T>(request: Request) => {
-  const { accessToken } = getTokens();
-  request.headers.set('Authorization', `Bearer ${accessToken}`);
-  return fetch(request).then((res) => {
-    try {
-      return handleResponse<T>(res);
-    } catch (error) {
-      if (error.statusCode === 401) {
+// No need to try renew token and retry request on login
+const shouldRetry = (url: string, error: any): boolean => {
```

Comment:
is `const shouldRetry = (url: string, status_code: string): boolean => {` a more suitable call signature?

Reply:
Yes I agree